csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -368,8 +353,14 @@ private protected async ValueTask WriteFlatBufferAsync(CancellationToken cancell
             await BaseStream.WriteAsync(segment, cancellationToken).ConfigureAwait(false);
         }
 
-        protected int CalculatePadding(int offset, int alignment = 8) =>
-            BitUtility.RoundUpToMultiplePowerOfTwo(offset, alignment) - offset;
+        protected int CalculatePadding(long offset, int alignment = 8)
+        {
+            long result = BitUtility.RoundUpToMultiplePowerOfTwo(offset, alignment) - offset;
+            checked
+            {
+                return (int)result;
+            }
```

Comment:
How come this uses checked with a cast, but earlier we use Convert.ToInt32?

Reply:
I think the other one was there before, and I just kept it. Do you have a preference on which to use? (I guess I would have picked checked with cast)

Reply:
> I guess I would have picked checked with cast

+1